catt/controllers.py

```diff
@@ -221,17 +231,29 @@ def get_data(self, name):
 
 class CastStatusListener:
     def __init__(self, app_id, active_app_id):
-        self.app_id = app_id
         self.app_ready = threading.Event()
+        self.set_app_id(app_id, active_app_id)
```

Comment:
What I wanted was this:

```
class CastStatusListener:
    def __init__(self, app_id, active_app_id):
        self.app_id = app_id
        self.app_ready = threading.Event()
        self.backdrop_ready = threading.Event()
        if active_app_id == app_id:
            self.app_ready.set()
        elif active_app_id == BACKDROP_APP_ID:
            self.backdrop_ready.set()

    def new_cast_status(self, status):
        if status.app_id == self.app_id:
            self.app_ready.set()
            self.backdrop_ready.clear()
        elif status.app_id == BACKDROP_APP_ID:
            self.backdrop_ready.set()
            self.app_ready.clear()
        else:
            self.app_ready.clear()
            self.backdrop_ready.clear()
```

You might want to know that after the listener has been registered in the constructor of the ```CastController``` baseclass, ```pychromecast``` automatically calls the ```new_cast_status``` method of the listener object whenever there is a change.
Also, ```threading.Event``` flags are initially set to false.